src/tools/rbd_mirror/instance_watcher/Types.h

```diff
@@ -44,7 +44,7 @@ struct ImagePayloadBase : public PayloadBase {
   std::string peer_mirror_uuid;
   std::string peer_image_id;
 
-  ImagePayloadBase() : PayloadBase() {
+  ImagePayloadBase() : PayloadBase(), global_image_id{""}, peer_mirror_uuid{""}, peer_image_id{""} {
```

Comment:
$ gcc --version
gcc (GCC) 7.1.1 20170503

You might be having an older version of gcc, in which it is not reproduced.

> Wont't it fix the warning if you do on the contrary, i.e. remove unnecessary PayloadBase() from the initialization

I don't think so, as the warnings points to the member variables initialization. If you want I can give it a try. Please let me know.

Thanks.